Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,12 +1,17 @@
 #include "BatteryStateOfChargeService.h"
-
+#include "BatteryData.h"
+#include <QTextStream>
+#include <cmath>
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const double CONVERT_MS_TO_HR = 3600000.0;
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
-: initialStateOfChargePercent_(initialStateOfChargePercent)
+: initialStateOfChargePercent_(initialStateOfChargePercent), current_(0.0), voltage_(0.0),
+time_(QTime(0,0,0,0)), lastCurrent_(0.0), lastTime_(QTime(0,0,0,0)), firstRun_(true),
+currentAmpHourUsed_((initialStateOfChargePercent_/100.0) * BATTERY_AMP_HOUR_CAPACITY)
```

Comment:
I didn't notice this before, but next time separate the list so that each variable as it's own line:
```
BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
: initialStateOfChargePercent_(initialStateOfChargePercent)
, current_(0.0)
, voltage_(0.0)
, time_(QTime(0,0,0,0))
, lastCurrent_(0.0)
, etc
```